src/operator/control_flow.cc

```diff
@@ -0,0 +1,428 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <mxnet/io.h>
+#include <mxnet/base.h>
+#include <mxnet/ndarray.h>
+#include <mxnet/operator.h>
+#include <mxnet/operator_util.h>
+#include <dmlc/logging.h>
+#include <dmlc/optional.h>
+#include "./operator_common.h"
+#include "./elemwise_op_common.h"
+#include "../imperative/imperative_utils.h"
+#include "./subgraph_op_common.h"
+
+namespace mxnet {
+namespace op {
+
+struct ForeachParam : public dmlc::Parameter<ForeachParam> {
+  int num_args;
+  int num_outputs;
+  int num_out_data;
+  nnvm::Tuple<dim_t> in_state_locs;
+  nnvm::Tuple<dim_t> in_data_locs;
```

Comment:
I think the scala failure is due to here `Tuple<dim_t>` somehow be translated to "tupleof<>" for function doc, but the newly introduced type-safe parsing cannot handle it: https://github.com/apache/incubator-mxnet/blob/master/scala-package/macros/src/main/scala/org/apache/mxnet/utils/CToScalaUtils.scala#L36

cc @lanking520 for visibility. also suggest to provide more detailed error msg (e.g., the operator name) to help debug more easily.

Reply:
@yzhliu Thank you for getting me involved. I confirm the problem on Scala build was caused by the API handling. The new Type tuple of <> cannot be interpreted on Scala side which cause the issue. Please feel free to add this case to the file that Yizhi indicated (place to the `Any` cases). I will also put a small patch on the Error message so developers can diagnose it easily.

Reply:
Is there any way of solving this problem quickly? I would like to have this PR merged soon

Reply:
@zheng-da Just add `| "tupleof<>"` at this [line](https://github.com/apache/incubator-mxnet/blob/master/scala-package/macros/src/main/scala/org/apache/mxnet/utils/CToScalaUtils.scala#L36), problem will be solved. I will keep supporting on this PR to make sure it meet your timeline.